app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Post;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //store the new post's information
+        $this->validate($request, [
+            'name' => 'required|max:20|min:1',
+            'description' => 'required|min:1|max:255',
+            'user_id' => 'required|integer'
```

Comment:
it is unusual to pass the `user_id` if the user is the currently authenticated user.

Rather, you can access the authenticated user via `$request->user()->id`. This removes an opportunity for someone to impersonate someone else by changing the hidden form input.